src/User/Filter/TwoFactorAuthenticationEnforceFilter.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the 2amigos/yii2-usuario project.
+ *
+ * (c) 2amigOS! <http://2amigos.us/>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Da\User\Filter;
+
+use Da\User\Model\User;
+use Da\User\Module;
+use Yii;
+use yii\base\ActionFilter;
+
+class TwoFactorAuthenticationEnforceFilter extends ActionFilter
+{
+    public function beforeAction($action)
+    {
+        /** @var Module $module */
+        $module = Yii::$app->getModule('user');
+
+        $enableTwoFactorAuthentication = $module->enableTwoFactorAuthentication;
+        // If enableTwoFactorAuthentication is set to false do nothing
+        if (!$enableTwoFactorAuthentication) {
+            return parent::beforeAction($action);
+        }
+
+        if (Yii::$app->user->isGuest) {
+            // Not our business
+            return parent::beforeAction($action);
+        }
+
+        /** @var User $identity */
+        $permissions =  $module->twoFactorAuthenticationForcedPermissions;
+        $identity = Yii::$app->user->identity;
+        foreach ( $permissions as $permission){
```

Comment:
Instead of a loop, you could use `AuthManager::getItemsByUser`, intersect it with `$permissions`. If the result is not empty, then the user has some permission which require 2FA